### PR TITLE
openssl@1.1: add linux-only changes

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by dawidd6** [comment]
Could you also put all the resources into one `on_linux do` block? I think it would be more readable and simple.

File: Formula/openssl@1.1.rb [hunk 1]
```diff
@@ -74,23 +93,44 @@ def install
...
     system "perl", "./Configure", *(configure_args + arch_args)
     system "make"
-    system "make", "test"
+    on_macos do
+      system "make", "test"
+    end
     system "make", "install", "MANDIR=#{man}", "MANSUFFIX=ssl"
+    # See https://github.com/Linuxbrew/homebrew-core/pull/8891
+    on_linux do
+      system "make", "test"
```
Comment (dawidd6) [inline Formula/openssl@1.1.rb:120]: How about moving `make test` after `make install` for both macOS and Linux? I guess it won't make a difference on macOS.

**Comment by SeekingMeaning** [comment]
> Could you also put all the resources into one `on_linux do` block? I think it would be more readable and simple.

I'm okay with either format, although we seem to have some formulae that have resources in `on_linux` and others that have `on_linux` in `resource` blocks. We probably should make them consistent

**Comment by dawidd6** [comment]
My opinion: more than 2 os-specific resources >> resources in on_os block.

**Comment by iMichka** [comment]
I rebased this and made the changes to the resources blocks.

We can skip the Mac CI as it will take around 28 hours to build.

**Review by dawidd6 — APPROVED** [review]
Looks okay

**Pull request merged**